tensorflow/core/kernels/reduction_ops_gpu_kernels.h

```diff
@@ -717,11 +728,11 @@ struct ReduceFunctor<GPUDevice, Eigen::internal::SumReducer<T>> {
   static void Reduce(OpKernelContext* ctx, OUT_T out, IN_T in,
                      const ReductionAxes& reduction_axes,
                      const Eigen::internal::SumReducer<T>& reducer) {
-    ReduceImpl<T, cub::Sum, T*, T*, ReductionAxes>(
+    ReduceImpl<T, Eigen::internal::scalar_sum_op<T>, T*, T*, ReductionAxes>(
```

Comment:
Are these changes for a different compiler problem?  Are other ops affected, or only sum?

Reply:
These correct a separate issue also related to reductions with std::complex types. cub::Sum relies on an operator overload for operator+, this is not available (in device code) for std::complex. This results in a compiler warning about calling "__host__" functions from "__host__ __device__ functions, and incorrect code.

Reply:
There are already workarounds for complex product and division; sum worked fine, but I guess is broken in nvcc 9?

Can you either change this fix to be consistent with the existing work arounds, or change the existing work arounds to this style?

Reply:
I made the code consistent with the existing workarounds.

cuda9 does have different behavior than cuda8. This may well be related to the same expt-relaxed-constexpr bug.